synapse/util/caches/deferred_cache.py

```diff
@@ -470,7 +470,7 @@ def __init__(self) -> None:
     def deferred(self, key: KT) -> "defer.Deferred[VT]":
         if not self._deferred:
             self._deferred = ObservableDeferred(defer.Deferred(), consumeErrors=True)
-        return self._deferred.observe().addCallback(lambda res: res.get(key))
+        return self._deferred.observe().addCallback(lambda res: res[key])
```

Comment:
I _think_ the point is that this lookup never fails---but please double check.

Reply:
I'm very not-confident about this change. I *think* you're correct, but I'm very lost in how this is used.

Reply:
Alright. I took another look at this and I agree that if this is *not* true then the internal machinery is broken. It'll raise errors anyway, so I think this is OK. (Otherwise you'd get back an unexpected `None`, which seems worse TBH)